source/adapters/hip/image.cpp

```diff
@@ -8,178 +8,1335 @@
+    CASE(UR_IMAGE_CHANNEL_TYPE_HALF_FLOAT, HIP_AD_FORMAT_HALF, 2, 0)
+    CASE(UR_IMAGE_CHANNEL_TYPE_UNSIGNED_INT32, HIP_AD_FORMAT_UNSIGNED_INT32, 4,
+         0)
+    CASE(UR_IMAGE_CHANNEL_TYPE_SIGNED_INT32, HIP_AD_FORMAT_SIGNED_INT32, 4, 0)
+    CASE(UR_IMAGE_CHANNEL_TYPE_FLOAT, HIP_AD_FORMAT_FLOAT, 4, 0)
+    CASE(UR_IMAGE_CHANNEL_TYPE_UNORM_INT8, HIP_AD_FORMAT_UNSIGNED_INT8, 1, 1)
+    CASE(UR_IMAGE_CHANNEL_TYPE_SNORM_INT8, HIP_AD_FORMAT_SIGNED_INT8, 1, 1)
+    CASE(UR_IMAGE_CHANNEL_TYPE_UNORM_INT16, HIP_AD_FORMAT_UNSIGNED_INT16, 2, 1)
+    CASE(UR_IMAGE_CHANNEL_TYPE_SNORM_INT16, HIP_AD_FORMAT_SIGNED_INT16, 2, 1)
+
+#undef CASE
+  default:
+    break;
+  }
+
+  if (return_hip_format) {
+    *return_hip_format = hip_format;
+  }
+  if (return_pixel_size_bytes) {
+    *return_pixel_size_bytes = pixel_size_bytes;
+  }
+  if (return_normalized_dtype_flag) {
+    *return_normalized_dtype_flag = normalized_dtype_flag;
+  }
+  return UR_RESULT_SUCCESS;
+}
+
+ur_result_t
+hipToUrImageChannelFormat(hipArray_Format hip_format,
+                          ur_image_channel_type_t *return_image_channel_type) {
+
+  switch (hip_format) {
+#define HIP_TO_UR_IMAGE_CHANNEL_TYPE(FROM, TO)                                 \
+  case FROM: {                                                                 \
+    *return_image_channel_type = TO;                                           \
+    return UR_RESULT_SUCCESS;                                                  \
+  }
+    HIP_TO_UR_IMAGE_CHANNEL_TYPE(HIP_AD_FORMAT_UNSIGNED_INT8,
+                                 UR_IMAGE_CHANNEL_TYPE_UNSIGNED_INT8);
+    HIP_TO_UR_IMAGE_CHANNEL_TYPE(HIP_AD_FORMAT_UNSIGNED_INT16,
+                                 UR_IMAGE_CHANNEL_TYPE_UNSIGNED_INT16);
+    HIP_TO_UR_IMAGE_CHANNEL_TYPE(HIP_AD_FORMAT_UNSIGNED_INT32,
+                                 UR_IMAGE_CHANNEL_TYPE_UNSIGNED_INT32);
+    HIP_TO_UR_IMAGE_CHANNEL_TYPE(HIP_AD_FORMAT_SIGNED_INT8,
+                                 UR_IMAGE_CHANNEL_TYPE_SIGNED_INT8);
+    HIP_TO_UR_IMAGE_CHANNEL_TYPE(HIP_AD_FORMAT_SIGNED_INT16,
+                                 UR_IMAGE_CHANNEL_TYPE_SIGNED_INT16);
+    HIP_TO_UR_IMAGE_CHANNEL_TYPE(HIP_AD_FORMAT_SIGNED_INT32,
+                                 UR_IMAGE_CHANNEL_TYPE_SIGNED_INT32);
+    HIP_TO_UR_IMAGE_CHANNEL_TYPE(HIP_AD_FORMAT_HALF,
+                                 UR_IMAGE_CHANNEL_TYPE_HALF_FLOAT);
+    HIP_TO_UR_IMAGE_CHANNEL_TYPE(HIP_AD_FORMAT_FLOAT,
+                                 UR_IMAGE_CHANNEL_TYPE_FLOAT);
+  default:
+    return UR_RESULT_ERROR_UNSUPPORTED_IMAGE_FORMAT;
+  }
+}
+
+ur_result_t urTextureCreate(ur_sampler_handle_t hSampler,
+                            const ur_image_desc_t *pImageDesc,
+                            const HIP_RESOURCE_DESC &ResourceDesc,
+                            const unsigned int normalized_dtype_flag,
+                            ur_exp_image_native_handle_t *phRetImage) {
+
+  try {
+    /// pi_sampler_properties
```

Comment:
This comment looks out of date? I wouldn't expect a `pi` here.

Reply:
Yeah the whole line here is irrelevant and out-of-date.